frontend/src/metabase/writeback/components/ActionCreator/QueryActionEditor.tsx

```diff
@@ -16,9 +16,14 @@ export function QueryActionEditor({
       <NativeQueryEditor
         query={question.query()}
         viewHeight="full"
-        setDatasetQuery={(newQuery: Query) =>
-          setQuestion(question.setQuery(newQuery))
-        }
+        setDatasetQuery={(newQuery: NativeQuery) => {
+          // we need to sync the query AND the template tags
+          setQuestion(
+            question
+              .setQuery(newQuery)
+              .setParameters(newQuery?.templateTagsWithoutSnippets()),
```

Comment:
This is the crucial bit. Honestly, I expected that updating the query would update the parameters which are derived from the template tags in the query 🤷

Reply:
paranoia 😨